src/payment/strategies/digitalriver/digitalriver-payment-strategy.spec.ts

```diff
@@ -390,6 +391,62 @@ describe('DigitalRiverPaymentStrategy', () => {
 
             return expect(promise).rejects.toBeInstanceOf(MissingDataError);
         });
+
+        describe('using vaulted cards', () => {
+            it('calls authenticateSource method when paying with vaulted instrument and 3DS is required', async () => {
+                jest.spyOn(paymentActionCreator, 'submitPayment')
+                    .mockReturnValueOnce(of(createErrorAction(PaymentActionType.SubmitPaymentFailed, getAdditionalActionError())));
+                jest.spyOn(paymentActionCreator, 'submitPayment')
+                    .mockReturnValueOnce(submitPaymentAction);
+                jest.spyOn(digitalRiverLoadResponse, 'authenticateSource').mockReturnValue(Promise.resolve({status: 'complete'}));
+
+                await strategy.initialize(options);
+                await strategy.execute(getOrderRequestBodyWithVaultedInstrument());
+
+                expect(paymentActionCreator.submitPayment).toHaveBeenCalledTimes(2);
+                expect(digitalRiverLoadResponse.authenticateSource).toHaveBeenCalled();
+            });
+
+            it('calls authenticateSource and authentication fails', async () => {
+                jest.spyOn(paymentActionCreator, 'submitPayment')
+                    .mockReturnValueOnce(of(createErrorAction(PaymentActionType.SubmitPaymentFailed, getAdditionalActionError())));
+                jest.spyOn(digitalRiverLoadResponse, 'authenticateSource').mockReturnValue(Promise.resolve({status: AuthenticationSourceStatus.failed}));
+
+                await strategy.initialize(options);
+                await expect(strategy.execute(getOrderRequestBodyWithVaultedInstrument())).rejects.toThrow();
+
+                expect(paymentActionCreator.submitPayment).toHaveBeenCalledTimes(1);
+                expect(digitalRiverLoadResponse.authenticateSource).toHaveBeenCalled();
```

Comment:
I dont see where you are expecting the fail

Reply:
Basically on line 416: `await expect(strategy.execute(getOrderRequestBodyWithVaultedInstrument())).rejects.toThrow();`

Reply:
that was my bad!